larpandoracontent/LArTrackShowerId/MvaPfoCharacterisationAlgorithm.cc

```diff
@@ -311,6 +318,14 @@ StatusCode MvaPfoCharacterisationAlgorithm<T>::ReadSettings(const TiXmlHandle xm
     PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle, "BdtName", m_mvaName));
     PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle, "SvmName", m_mvaName));
     PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle, "MvaName", m_mvaName));
+    PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle,
+        "WriteToTree", m_writeToTree)); // added by Mousam
+
+    PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle,
+        "OutputTree", m_treeName)); // added by Mousam
+    
+    PANDORA_RETURN_RESULT_IF_AND_IF(STATUS_CODE_SUCCESS, STATUS_CODE_NOT_FOUND, !=, XmlHelper::ReadValue(xmlHandle,
+        "OutputFile", m_fileName)); //added by Mousam
```

Comment:
If WriteToTree is set should probably add a conditional block that requires the presence of the output tree and filenames, as in this form, if these are missing in the XML file then they will default to being empty without any complaint from ReadSettings.

Reply:
Only cosmetic, but I'd probably remove the white lines too between the various PANDORA_RETURN_RESULT_IF_AND_IF, for consistency.

Reply:
@AndyChappell Dont think that is require as this can be changed in the neutrino settings file, right? If you dont want to produce root file then you can set <WriteToTree>to "false". I thought that was enough. If you do that, then there shouldn't be any errors.

Reply:
@MariaBrigida Done!

Reply:
You can set the flag to not produce the root file, but my concern here is that if you do set the flag to produce the root file, you could in principle forget to set the tree name and filename in XML, and as these don't appear to be set anywhere by default, I think you would run into trouble when trying to write the output file - in which case I think it's worth enforcing the presence of the name variables in the event that the output flag is set to true.

Reply:
Maybe I am not getting what you are saying. I was under the assumption that <OutputTree> and <OutputFile> flags were available in the neutrino settings file too (https://github.com/MoteyMousam/LArReco/blob/0d59898780377b86dc604981f63cd58c21e4e96f/settings/PandoraSettings_Neutrino_DUNEFD.xml#L371). I could add something similar (i think) in line 48 of the algorithm (https://github.com/MoteyMousam/LArContent/blob/ce0f96db61c65d962f3169d04717cdac6785de58/larpandoracontent/LArTrackShowerId/MvaPfoCharacterisationAlgorithm.cc#L48). Would it not be more cleaner to use the neutrino settings xml file instead of hard coding the requirement into the alg?

Reply:
@AndyChappell done